### ISSUE TITLE
Better UI for creating/editing libraries

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#6684]
Following #6518/#5495.

The idea is that we want to make it more explicit how to handle libraries. In general, those will be few, typically a couple:

- global library (`_`)
- app library

Steps:

- [ ] split "New Form"/"New Library
- [ ] validate `_`/`library`
- [ ] could be a new tab in the Form Settings dialog?
    -  but then would need ability to switch from form to library? or not?
- [ ] ~~create new template for "Library"~~ Or separate tab? Or checkbox?
- [ ] when editing, must set form name to `library` automatically
- [ ] allow importing Global Library from resources if the library does not exist in Form Builder or published (see #5495)
    - [ ] import UI
    - [ ] ship `_` library in resources

**Comment by NightingaleCrypter** [comment]
Hello Mr. Erik,
if I might drop suggestion, maybe some "goto-icon" / "init-app-library-icon" / "init-global-library-icon" inside toolbox to navigate towards libraries? That's where we users click to fetch defined sections in templates, so feels good UX that icon to create library or open builder-edit mode for it would be quite intuitive (instead of "could be a new tab in the Form Settings dialog").
Just a thought, don't hate me.
Kind regards,
Marcin

**Comment by avernet** [comment]
@NightingaleCrypter Yes, I think what you're suggesting makes a lot of sense. Here are a couple of mock-ups.

- The first one uses an icon to the right of the library title and shows empty libraries with an indicator. There should also be a tooltip on the create/edit library icon.
- The second mock-up uses a regular button to create or edit the library and uses different icons to distinguish it from the buttons to add a section template.

Number two looks easier to implement from a UI perspective, but I prefer number one.

<img width="203" alt="Image" src="https://github.com/user-attachments/assets/05f7ee02-50e7-41b3-83c0-ebd3dd21d8ff" />
<img width="203" alt="Image" src="https://github.com/user-attachments/assets/d817fac9-e445-405a-b77a-c34015a8ebf7" />

[library-1.html](https://github.com/user-attachments/files/28121410/library-1.html)
[library-2.html](https://github.com/user-attachments/files/28121426/library-2.html)

**Comment by NightingaleCrypter** [comment]
1. It is natural UX design.
2. It looks actually okayish when creating library, but editing library looks a bit lost in the context. But something is always better than nothing. In one of the projects that was using self-managed Orbeon I saw that they even ditched random UUID to something like "{appname}{formname}" on form builders so they can quickly rename second part to access library.

Anyway looking forward to it, mockups looks awesome. (Just as Your whole app)

**Comment by avernet** [comment]
- [x] Check if need to do something  for permissions.
    - Nothing to do: search take the user's permission into account.
- [x] Do the search on click, not ahead of time.
    - The idea would be to avoid always doing 3 searches every time we load a form.
    - For a given library type (`orbeon`, `_`, app):
        - If `orbeon` and there is no published library: don't show anything in the sidebar.
        - Otherwise: link triggers a search, based on which we display a dialog for users, depending on the search results:
            - No results: link to create a library
            - One result: link to open that library
            - Multiple result: link to the summary page with request parameter so the user can select the library to open
